### PR TITLE
[DEV-1631] Add slug custom uniq validator

### PR DESCRIPTION
**Pull request by MarcoPonchia**
depends on: #857 

#### List of Changes

Add slug custom uniq validator

#### Motivation and Context

Webinars must have a unique slug for the same language in Strapi to be correctly fetch in NextJS once multilingual will be supported

#### How Has This Been Tested?

Editor see an error if try to create or update webinar if slug already exist for another webinar in same language

#### Screenshots (if appropriate):
image

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/strapi-cms/src/api/webinar/content-types/webinar/lifecycles.ts [hunk 1]
```diff
@@ -20,22 +25,68 @@ const validateDates = (event: IWebinarEvent) => {
...
+    // eslint-disable-next-line functional/no-throw-statements
     throw new errors.ApplicationError(
-      'Both start and end dates must be provided, or none should be set'
+      "Both start and end dates must be provided, or none should be set"
     );
   }
 
   if (startDateTime && endDateTime && endDateTime <= startDateTime) {
-    throw new errors.ApplicationError('End date must be after start date');
+    throw new errors.ApplicationError("End date must be after start date");
```
Comment (marcobottaro) [inline apps/strapi-cms/src/api/webinar/content-types/webinar/lifecycles.ts:35]:
```suggestion
    throw new errors.ApplicationError('End date must be after start date');
```

File: apps/strapi-cms/src/api/webinar/content-types/webinar/lifecycles.ts [hunk 2]
```diff
@@ -1,17 +1,22 @@
 // The only way to make this code work is to do side effects, and throwing exceptions and creating functions
 /*eslint-disable*/
-import { errors } from '@strapi/utils';
+import { errors } from "@strapi/utils";
```
Comment (marcobottaro) [inline apps/strapi-cms/src/api/webinar/content-types/webinar/lifecycles.ts:3]:
```suggestion
import { errors } from '@strapi/utils';
```

File: apps/strapi-cms/src/api/webinar/content-types/webinar/lifecycles.ts [hunk 3]
```diff
@@ -20,22 +25,68 @@ const validateDates = (event: IWebinarEvent) => {
...
   }
+  return true;
+};
+
+const getLocale = async (event: IWebinarEvent): Promise<string> => {
+  if (event.params.data.locale) {
+    return event.params.data.locale;
+  } else {
+    const webinar: IWebinar | undefined = await strapi
+      .service("api::webinar.webinar")
```
Comment (marcobottaro) [inline apps/strapi-cms/src/api/webinar/content-types/webinar/lifecycles.ts:45]:
```suggestion
      .service('api::webinar.webinar')
```

File: apps/strapi-cms/src/api/webinar/content-types/webinar/lifecycles.ts [hunk 4]
```diff
@@ -20,22 +25,68 @@ const validateDates = (event: IWebinarEvent) => {
...
+      .findOne(event.params.data.id);
+    return webinar?.locale || "it";
```
Comment (marcobottaro) [inline apps/strapi-cms/src/api/webinar/content-types/webinar/lifecycles.ts:47]:
```suggestion
    return webinar?.locale || 'it';
```

File: apps/strapi-cms/src/api/webinar/content-types/webinar/lifecycles.ts [hunk 5]
```diff
@@ -20,22 +25,68 @@ const validateDates = (event: IWebinarEvent) => {
...
+  if (event.params.data.id) {
+    query.where = {
+      id: {
+        $notIn: [event.params.data.id],
+      },
+      ...query.where,
+    };
+  }
+  const webinarWithSameSlug: IWebinar | undefined = await strapi.db
+    .query("api::webinar.webinar")
```
Comment (marcobottaro) [inline apps/strapi-cms/src/api/webinar/content-types/webinar/lifecycles.ts:71]:
```suggestion
    .query('api::webinar.webinar')
```

File: apps/strapi-cms/src/api/webinar/content-types/webinar/lifecycles.ts [hunk 6]
```diff
@@ -20,22 +25,68 @@ const validateDates = (event: IWebinarEvent) => {
...
+    .findOne(query);
+
+  if (webinarWithSameSlug) {
+    // eslint-disable-next-line functional/no-throw-statements
+    throw new errors.ApplicationError(
+      "Webinar with the same slug already exists for the current locale"
```
Comment (marcobottaro) [inline apps/strapi-cms/src/api/webinar/content-types/webinar/lifecycles.ts:77]:
```suggestion
      'A Webinar with the same slug already exists for the current locale'
```

**Review by marcobottaro — CHANGES_REQUESTED** [review]

**Comment by marcobottaro** [comment]
I see the following error when trying to publish a Webinar:
```error: Undefined binding(s) detected when compiling WHERE. Undefined column(s): [t0.id] query: where `t0`.`id` = ?
Error: Undefined binding(s) detected when compiling WHERE. Undefined column(s): [t0.id] query: where `t0`.`id` = ?
    at QueryCompiler_SQLite3.toSQL (/home/marco/Projects/pagoPA/backend/developer-portal/node_modules/knex/lib/query/querycompiler.js:112:13)
    at compileCallback (/home/marco/Projects/pagoPA/backend/developer-portal/node_modules/knex/lib/formatter/formatterUtils.js:13:19)
    at rawOrFn (/home/marco/Projects/pagoPA/backend/developer-portal/node_modules/knex/lib/formatter/wrappingFormatter.js:225:7)
    at QueryCompiler_SQLite3.whereWrapped (/home/marco/Projects/pagoPA/backend/developer-portal/node_modules/knex/lib/query/querycompiler.js:1095:17)
    at QueryCompiler_SQLite3.where (/home/marco/Projects/pagoPA/backend/developer-portal/node_modules/knex/lib/query/querycompiler.js:584:34)
    at /home/marco/Projects/pagoPA/backend/developer-portal/node_modules/knex/lib/query/querycompiler.js:135:40
    at Array.forEach (<anonymous>)
    at QueryCompiler_SQLite3.select (/home/marco/Projects/pagoPA/backend/developer-portal/node_modules/knex/lib/query/querycompiler.js:134:16)
    at QueryCompiler_SQLite3.toSQL (/home/marco/Projects/pagoPA/backend/developer-portal/node_modules/knex/lib/query/querycompiler.js:75:29)
    at QueryBuilder_SQLite3.toSQL (/home/marco/Projects/pagoPA/backend/developer-portal/node_modules/knex/lib/query/querybuilder.js:84:44)```

File: apps/strapi-cms/src/api/webinar/content-types/webinar/lifecycles.ts
[the remark this answers is not in the document]
Comment (MarcoPonchia) [reply apps/strapi-cms/src/api/webinar/content-types/webinar/lifecycles.ts:1]: Adding more specific rules would result in having almost double the lines in this file and would make it difficult to read. I preferred to solve [this task](https://pagopa.atlassian.net/jira/software/projects/DEV/boards/280/backlog?selectedIssue=DEV-1594&text=lint) and make this PR depend on [this one](https://github.com/pagopa/developer-portal/pull/857), where I added a specific eslint configuration for Strapi.

**Comment by marcobottaro** [comment]
LGTM :+1:

**Review by marcobottaro — APPROVED** [review]

**Pull request merged**